src/utilities/ConvectiveOperator.cpp

```diff
@@ -80,6 +80,15 @@ ConvectiveOperator::apply(SAMRAIVectorReal<NDIM, double>& x, SAMRAIVectorReal<ND
     return;
 } // apply
 
+void
+ConvectiveOperator::applyConvectiveOperator(int /*Q_idx*/, int /*P_idx*/, int /*N_idx*/)
+{
+    TBOX_ERROR(
+        "ConvectiveOperator::applyConvectiveOperator is a base class and the implementation is provided in "
+        "AdvDiffCUIConservativeConvectiveOperator class.\n");
+    return;
+} // applyConvectiveOperator
```

Comment:
Is there a plan to implement this function in all of the derived classes? If there isn't a way to do so then this function should just be in the derived class.

Reply:
I think this is implemented in a derived class. @Raamkrishnan can point out to the exact case.

Reply:
I think Boyce suggested to have it in the base `ConvectiveOperator` class. We didn't have it before. It was done (with some effort) after the suggestion.